LoRaEngine/LoraKeysManagerFacade/FacadeStartup.cs

```diff
@@ -67,21 +73,60 @@ public override void Configure(IFunctionsHostBuilder builder)
                     .AddSingleton<IFunctionBundlerExecutionItem, ADRExecutionItem>()
                     .AddSingleton<IFunctionBundlerExecutionItem, PreferredGatewayExecutionItem>()
                     .AddSingleton<LoRaDevAddrCache>();
+
+
+            if (configHandler.DeviceRegistryMode == DeviceRegistryMode.IoTHub)
+            {
+                var regex = new Regex("HostName=(.*[.]azure[-]devices[.]net);");
```

Comment:
Improvements: 
```suggestion
                var regex = new Regex(@"\bHostName=(.+\.azure-devices\.net);", RegexOptions.IgnoreCase);
```

- Make the C# string _verbatim_ with `@`
- Require `HostName` match on a word boundary.
- `*` should be `+` so it does not match empty
- Escape `.` with `\.`
- `-` doesn't need to be escaped
- Use `RegexOptions.IgnoreCase | RegexOptions.CultureInvariant` for case-insensitive matches in the invariant culture.

Reply:
Also don't you think it'll be simpler and safer to use [`IotHubConnectionStringBuilder.Create`](https://docs.microsoft.com/en-us/dotnet/api/microsoft.azure.devices.client.iothubconnectionstringbuilder.create?view=azure-dotnet#Microsoft_Azure_Devices_Client_IotHubConnectionStringBuilder_Create_System_String_) to parse the string and then access [its `HostName` property](https://docs.microsoft.com/en-us/dotnet/api/microsoft.azure.devices.client.iothubconnectionstringbuilder.hostname?view=azure-dotnet#Microsoft_Azure_Devices_Client_IotHubConnectionStringBuilder_HostName)?